lightning/src/ln/channel.rs

```diff
@@ -2585,7 +2585,10 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 			next_remote_commitment_tx_fee_info_cached: Mutex::new(None),
 
 			channel_transaction_parameters: ChannelTransactionParameters {
-				holder_pubkeys: pubkeys,
+				holder_pubkeys: HolderChannelPublicKeys::new(
+					pubkeys.funding_pubkey, pubkeys.revocation_basepoint, pubkeys.payment_point,
+					pubkeys.delayed_payment_basepoint, pubkeys.htlc_basepoint, None, &secp_ctx,
+				),
```

Comment:
Consider implementing `From` instead.

Reply:
`impl From<ChannelPublicKeys> for HolderChannelPublicKeys` since `pubkeys` isn't used anywhere else in the method, and there's no need to pass in `&secp_ctx` when we pass `None` to that constuctor.